### PR TITLE
EVG-5887 Fix up remote queue group

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: queue/group_remote.go [hunk 1]
```diff
@@ -108,7 +116,7 @@ func NewRemoteQueueGroup(ctx context.Context, opts RemoteQueueGroupOptions) (amb
 				case <-ctx.Done():
 					return
 				case <-ticker.C:
-					_ = g.Prune(ctx)
+					grip.Error(errors.Wrap(g.Prune(pruneCtx), "problem pruning remote queue group database"))
```
Comment (tychoish) [inline queue/group_remote.go]: why not collect this and propogate out? catchers are thread safe. 

also message.WrapError() is probably also good/better?

File: queue/group_remote.go [hunk 2]
```diff
@@ -250,21 +261,24 @@ func (g *remoteQueueGroup) Prune(ctx context.Context) error {
...
-	for i, coll := range colls {
+	collsToCheck := []string{}
+	for _, coll := range colls {
 		// This is an optimization. If we've added to the queue recently enough, there's no
 		// need to query its contents, since it cannot be old enough to prune.
 		if t, ok := g.ttlMap[g.idFromCollection(coll)]; ok && time.Since(t) < g.ttl {
-			g.remove(colls, i)
+			// pass
+		} else {
+			collsToCheck = append(collsToCheck, coll)
```
Comment (tychoish) [inline queue/group_remote.go]: invert the logic to avoid the else?

File: queue/group_remote.go [hunk 3]
```diff
@@ -250,21 +261,24 @@ func (g *remoteQueueGroup) Prune(ctx context.Context) error {
...
 	catcher := grip.NewBasicCatcher()
 	wg := &sync.WaitGroup{}
-	collsDeleteChan := make(chan string, len(colls))
+	collsDeleteChan := make(chan string, len(collsToCheck))
 	collsDropChan := make(chan string)
 
 	go func() {
 		defer recovery.LogStackTraceAndContinue("panic in pruning collections")
-		for _, coll := range colls {
+		for _, coll := range collsToCheck {
```
Comment (tychoish) [inline queue/group_remote.go]: given that this collsDropChan is buffered, do we need to do this in a thread?

File: queue/group_remote.go [hunk 4]
```diff
@@ -286,7 +300,7 @@ func (g *remoteQueueGroup) Prune(ctx context.Context) error {
 					catcher.Add(err)
 					return
 				}
-				if err := one.Decode(struct{}{}); err != nil {
+				if err := one.Decode(&struct{}{}); err != nil {
```
Comment (tychoish) [inline queue/group_remote.go]: given that we don't care about the result, shouldn't we just be doing counts here?

File: queue/group_remote.go (hunk 1 above)
Comment (bsamek) [reply queue/group_remote.go]: This goroutine keeps running after the constructor exits.

File: queue/group_remote.go (hunk 2 above)
Comment (bsamek) [reply queue/group_remote.go]: done

File: queue/group_remote.go (hunk 3 above)
Comment (bsamek) [reply queue/group_remote.go]: done

File: queue/group_remote.go (hunk 4 above)
Comment (bsamek) [reply queue/group_remote.go]: done

**Pull request merged**